File: src/main/java/org/dicio/skill/standard/InputWordRange.java
package org.dicio.skill.standard;

public class InputWordRange {
    private int from;
    private int to;

    /**
     * Deep copy constructor
     */
    InputWordRange(final InputWordRange other) {
        from = other.from;
        to = other.to;
    }

    /**
     * @param from the index of the first element of the range
     * @param to the index of one past the last element of the range
     */
    public InputWordRange(final int from, final int to) {
        this.from = from;
        this.to = to;
    }

    @Override
    public String toString() {
        return "[" + from + "," + to + ")";
    }

    /**
     * @return the index of the first element of the range
     */
    public int from() {
        return from;
    }

    /**
     * @return the index of one past the last element of the range
     */
    public int to() {
        return to;
    }
}

File: skill/src/test/java/org/dicio/skill/old_standard_impl/InputWordRange.kt
package org.dicio.skill.old_standard_impl

class InputWordRange {
    private var from: Int
    private var to: Int

    /**
     * Deep copy constructor
     */
    internal constructor(other: InputWordRange) {
        from = other.from
        to = other.to
    }

    /**
     * @param from the index of the first element of the range
     * @param to the index of one past the last element of the range
     */
    constructor(from: Int, to: Int) {
        this.from = from
        this.to = to
    }

    override fun toString(): String {
        return "[$from,$to)"
    }

    /**
     * @return the index of the first element of the range
     */
    fun from(): Int {
        return from
    }

    /**
     * @return the index of one past the last element of the range
     */
    fun to(): Int {
        return to
    }
}
